src/main/java/spoon/support/visitor/MethodTypingContext.java

```diff
@@ -249,6 +250,27 @@ private boolean isSameMethodFormalTypeParameter(CtTypeParameter scopeParam, CtTy
 				return idxOfScopeBoundTypeParam == idxOfSuperBoundTypeParam;
 			}
 		}
+		if (scopeBound.getActualTypeArguments().size() != superBound.getActualTypeArguments().size()) {
+			return false;
+		}
+
+		for (int i = 0; i < scopeBound.getActualTypeArguments().size(); i++) {
+			CtTypeReference scopeBoundATArg = scopeBound.getActualTypeArguments().get(i);
+			CtTypeReference superBoundATArg = superBound.getActualTypeArguments().get(i);
+
+			if (scopeBoundATArg instanceof CtWildcardReference && superBoundATArg instanceof CtWildcardReference) {
+				CtWildcardReference scopeBoundWildcard = (CtWildcardReference) scopeBoundATArg;
+
+				// we are in a case where we compare Thing<?> with Thing<? super X>
+				// here we choose to not care about X to avoid recursive calls, for example in the case of:
+				// T extends Comparable<? super T>
+				if (scopeBoundWildcard.getBoundingType().equals(scopeMethod.getFactory().Type().getDefaultBoundingType())) {
```

Comment:
May this happen too?
`T extends Comparable<? super List<? extends T>>`
I guess it will end wit stackoverflow too.

Anyway your solution is good fast fix. My example is quite wild and it will need some time until somebody come to such problem again.